frontend/public/components/utils/request-size-input.tsx

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-undef, no-unused-vars */
+import * as React from 'react';
+import { Dropdown } from '.';
+
+export class RequestSizeInput extends React.Component<RequestSizeInputProps> {
+
+  onChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    this.props.onChange(event);
+  };
+
+  onDropdownChange = event => {
+    let obj = { currentTarget: { name: `${this.props.name}Unit`, value: event } };
+    this.props.onChange(obj);
```

Comment:
My handler expects an object with name and value, but the dropdown part of the RSI component only returns a value.

Reply:
> My handler expects an object with name and value, but the dropdown part of the RSI component only returns a value.

Right, I'm suggesting we change the handler instead of creating a fake event :)

Reply:
Well, I currently on pass one handler into the RSI component.  How will I know which part of the RSI component (dropdown or input) is sending me the change.  I will have to put special logic in the parent handler to detect if the event is a string or an object.  That seems more wonky than just having the parent always expect an object with name and value.

Reply:
I'm saying never pass the event. Only pass the value. Does the parent need the event?